src/sinks/elasticsearch/config.rs

```diff
@@ -43,10 +45,11 @@ pub const DATA_STREAM_TIMESTAMP_KEY: &str = "@timestamp";
 #[derive(Clone, Debug, Default)]
 #[serde(deny_unknown_fields)]
 pub struct ElasticsearchConfig {
-    /// The Elasticsearch endpoint to send logs to.
+    /// The Elasticsearch endpoint/endpoints to send logs to.
     ///
     /// This should be the full URL as shown in the example.
-    pub endpoint: String,
+    #[configurable(derived)]
+    pub endpoint: OneOrMany<String>,
```

Comment:
Maybe others have strong opinions on this, but keeping the name singular `endpoint` seems odd to me.
I'd almost prefer to have it be `endpoints` and the type be `Vec<String>`, then it is more clear :shrug: . I realize that would require modifications to existing configs, so perhaps that is not desirable?

Reply:
I'd also 👍 a deprecation of the existing and move to `endpoints: Vec<String>` here.

Reply:
`endpoints` naming would be preferable. 

A middle ground would be to leave `OneOrMany<String>` and add an alias `endpoint`. We would have preferred naming and would maintain backward compatibility.

Reply:
I'm also a fan of introducing a new mutually exclusive `endpoints` option. This will make documenting it easier too per https://github.com/vectordotdev/vector/pull/14088#discussion_r971124740

Reply:
Then I'll add `endpoints` option that is mutually exclusive, and deprecate `endpoint` option.